src/bun.js/bindings/bindings.zig

```diff
@@ -3736,11 +3742,19 @@ pub const JSValueReprInt = i64;
 /// ABI-compatible with EncodedJSValue
 /// In the future, this type will exclude `zero`, encoding it as `error.JSError` instead.
 pub const JSValue = enum(i64) {
-    zero = 0,
     undefined = 0xa,
     null = 0x2,
     true = FFI.TrueI64,
     false = 0x6,
+
+    /// Typically means an exception was thrown.
+    zero = 0,
+
+    /// JSValue::ValueDeleted
+    ///
+    /// Deleted is a special encoding used in JSC hash map internals used for
+    /// the null state. It is re-used here for encoding the "not present" state.
+    property_does_not_exist_on_object = 0x4,
```

Comment:
this was very very intentionally not added to JSValue.

> It is re-used here for encoding the "not present" state.
this comment is now out of date because "here" doesn't mean anything